packages/sdk/src/sdk/api/events/types.ts

```diff
@@ -0,0 +1,38 @@
+import { z } from 'zod'
+
+import { EventEventTypeEnum, EventEntityTypeEnum } from '../generated/default'
+
+// Base schema for event metadata
+export const EventMetadataSchema = z.object({
+  userId: z.number(),
+  eventId: z.number(),
+  eventType: z.nativeEnum(EventEventTypeEnum).optional(),
+  entityType: z.nativeEnum(EventEntityTypeEnum).optional(),
+  entityId: z.number().optional(),
+  endDate: z.string().optional(), // ISO format date string
```

Comment:
If this is common for all our other metadata types, I'd leave it. Otherwise I'd just make this a date and we can ISO string it internally. Too easy to mess this up and pass it in the wrong format, and we aren't validating that it's a valid data string.